### PR TITLE
store/tikv: remove tidb/metrics from store/tikv

### PR DESCRIPTION
**Pull request by aierui**
### What problem does this PR solve?

Remove store/tikv's dependency on  tidb/metrics
Part of #22513 

### What is changed and how it works?

What's Changed:

store/tikv: remove tidb/metrics from store/tikv/client.go,store/tikv/client_batch.go.

### Check List 

Tests 

- Unit test

### Release note 
- No release note

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -139,14 +168,14 @@ func RegisterMetrics() {
...
 	prometheus.MustRegister(MaxProcs)
 	prometheus.MustRegister(GOGC)
 	prometheus.MustRegister(ConnIdleDurationHistogram)
 	prometheus.MustRegister(ServerInfo)
 	prometheus.MustRegister(TokenGauge)
 	prometheus.MustRegister(ConfigStatus)
 
-	tikvmetrics.InitMetrics("tidb", "tikvclient")
+	tikvmetrics.InitMetrics(TiDB, TiKVClient)
+	tikvmetrics.TiKVPanicCounter = initMetric(panicCounter.resetMetricDomain(TiDB, TiKVClient)) // reset tidb metrics for tikv metrics
```
Comment (disksing) [inline metrics/metrics.go:179]: Why do we need to recreate the counterVec?

Comment (aierui) [reply metrics/metrics.go:179]: Because `subsystem` param of the two Components(TiDB-Server,TiKV-Client-Go).

File: metrics/metrics.go [hunk 2]
```diff
@@ -46,11 +46,21 @@ const (
...
 
+	TiDB         = "tidb"
 	LabelScope   = "scope"
 	ScopeGlobal  = "global"
 	ScopeSession = "session"
+	Server       = "server"
+	TiKVClient   = "tikvclient"
 )
 
+type metrics struct {
```
Comment (disksing) [inline metrics/metrics.go:57]: no use?

Comment (aierui) [reply metrics/metrics.go]: yes, I will delete it immediately.

**Review by disksing — APPROVED** [review]

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**